### PR TITLE
PWGHF: save info isGlobalTrakWoDca for pion 

### PR DESCRIPTION
**Pull request by ZFederica**
In the toXiPi workflow, save info isGlobalTrackWoDca for pion <-- charm (offline analysis purposes)

### CODE REVIEW

**Review by fcatalan92 — COMMENTED** [review]
Hi @ZFederica, in general the PR is fine to me. I have just a question: is adding a column in the candidate table necessary? It looks to me you could use `hf_cand_toxipi::PiFromCharmBaryonId` to retrieve the same information in the tree creator.

**Comment by ZFederica** [comment]
Hi @fcatalan92, sure, I removed it from the candidateCreator

File: PWGHF/TableProducer/treeCreatorToXiPi.cxx [hunk 1]
```diff
@@ -181,7 +182,7 @@ struct HfTreeCreatorToXiPi {
   }
 
   template <typename T>
-  void fillCandidate(const T& candidate, int8_t flagMc, int8_t debugMc, int8_t originMc)
+  void fillCandidate(const T& candidate, bool IsPionGlbTrkWoDca, int8_t flagMc, int8_t debugMc, int8_t originMc)
```
Comment (fcatalan92) [inline PWGHF/TableProducer/treeCreatorToXiPi.cxx:185]:
```suggestion
  void fillCandidate(const T& candidate, int8_t flagMc, int8_t debugMc, int8_t originMc)
```

File: PWGHF/TableProducer/treeCreatorToXiPi.cxx [hunk 2]
```diff
@@ -266,6 +267,7 @@ struct HfTreeCreatorToXiPi {
       candidate.impactParCascXY() / candidate.errImpactParCascXY(),
       candidate.impactParPiFromCharmBaryonXY() / candidate.errImpactParPiFromCharmBaryonXY(),
       candidate.decLenCharmBaryon() / candidate.errorDecayLengthCharmBaryon(),
+      IsPionGlbTrkWoDca,
```
Comment (fcatalan92) [inline PWGHF/TableProducer/treeCreatorToXiPi.cxx:270]:
```suggestion
      candidate.template PiFromCharmBaryon_as<aod::TrackSelection>().isGlobalTrackWoDCA(),
```

File: PWGHF/TableProducer/treeCreatorToXiPi.cxx [hunk 3]
```diff
@@ -288,13 +290,18 @@ struct HfTreeCreatorToXiPi {
...
     // Filling candidate properties
     rowCandidateFull.reserve(candidates.size());
     for (const auto& candidate : candidates) {
-      fillCandidate(candidate, -7, -7, RecoDecay::OriginType::None);
+
+      // check if pion track satisfies globalTrackWoDca selections
+      auto trackPionSelInfo = candidate.PiFromCharmBaryon_as<aod::TrackSelection>();
+      bool isPiGlbTrkWoDca = trackPionSelInfo.isGlobalTrackWoDCA();
+
+      fillCandidate(candidate, isPiGlbTrkWoDca, -7, -7, RecoDecay::OriginType::None);
```
Comment (fcatalan92) [inline PWGHF/TableProducer/treeCreatorToXiPi.cxx:304]:
```suggestion
      fillCandidate(candidate, -7, -7, RecoDecay::OriginType::None);
```

File: PWGHF/TableProducer/treeCreatorToXiPi.cxx [hunk 4]
```diff
@@ -305,7 +312,12 @@ struct HfTreeCreatorToXiPi {
     // Filling candidate properties
     rowCandidateFull.reserve(candidates.size());
     for (const auto& candidate : candidates) {
-      fillCandidate(candidate, candidate.flagMcMatchRec(), candidate.debugMcRec(), candidate.originRec());
+
+      // check if pion track satisfies globalTrackWoDca selections
+      auto trackPionSelInfo = candidate.PiFromCharmBaryon_as<aod::TrackSelection>();
+      bool isPiGlbTrkWoDca = trackPionSelInfo.isGlobalTrackWoDCA();
+
+      fillCandidate(candidate, isPiGlbTrkWoDca, candidate.flagMcMatchRec(), candidate.debugMcRec(), candidate.originRec());
```
Comment (fcatalan92) [inline PWGHF/TableProducer/treeCreatorToXiPi.cxx:320]:
```suggestion
      fillCandidate(candidate, candidate.flagMcMatchRec(), candidate.debugMcRec(), candidate.originRec());
```

**Review by fcatalan92 — COMMENTED** [review]
Out of curiosity, could you check if the following approach works? I made suggestions for all the necessary changes.

File: PWGHF/TableProducer/treeCreatorToXiPi.cxx (hunk 2 above)
Comment (fcatalan92) [reply PWGHF/TableProducer/treeCreatorToXiPi.cxx:270]: This should fix the compilation error too

**Comment by ZFederica** [comment]
Thanks, I missed the capital letter

**Review by fcatalan92 — APPROVED** [review]

**Pull request merged**